infra/launch.py

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/python
+
+import argparse
+import getpass
+import subprocess
+import time
+
+from infra import push_docker
+
+from infra.helpers import cli
+
+
+def setup_vm_docker(tpu_name, zone, docker_base_image):
+    cli.tpu_ssh(
+        tpu_name,
+        zone,
+        "sudo",
+        "usermod",
+        "-aG",
+        "docker",
+        getpass.getuser(),
+    )
+
+    cli.tpu_ssh(
+        tpu_name,
+        zone,
+        "docker",
+        "pull",
+        docker_base_image,
+    )
+
+    cli.tpu_ssh(tpu_name, zone, "docker", "volume", "create", "--driver=local", "levanter")
+
+
+def list_tpus(zone):
+    tpus = subprocess.check_output(
+        [
+            "gcloud",
+            "alpha",
+            "compute",
+            "tpus",
+            "tpu-vm",
+            "list",
+        ]
+    )
+    rows = tpus.decode("utf-8").split("\n")
+    header = rows[0].split()
+    tpus = []
+    for row in rows[1:]:
+        if row:
+            tpus.append(dict(zip(header, row.split())))
+    return tpus
+
+
+def start_tpu_vm(
+    tpu_name, *, tpu_type, preemptible, version, zone, autodelete, project, docker_repository, docker_base_image
+):
+    tpu_exists = any([tpu["NAME"] == tpu_name for tpu in list_tpus(zone)])
+    if tpu_exists:
+        if not autodelete:
+            print("TPU already exists and autodelete is false, leaving it as is.")
+            return
+
+        print("TPU already exists, deleting...")
+        cli.run_command(
+            "gcloud",
+            "alpha",
+            "compute",
+            "tpus",
+            "delete",
+            "--quiet",
+            f"--zone={zone}",
+            tpu_name,
+        )
+
+    print(f"Creating new TPU {tpu_name} in {zone} of type {tpu_type}...")
+    cli.run_command(
+        "gcloud",
+        "alpha",
+        "compute",
+        "tpus",
+        "tpu-vm",
+        "create",
+        tpu_name,
+        f"--accelerator-type={tpu_type}",
+        f"--version={version}",
+        "--zone=" + zone,
+        "--preemptible" if preemptible else "",
+        "--quiet",
+    )
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    config = cli.load_config()
+
+    cli.add_arg(parser, config, ["--autodelete"], default=False, action="store_true")
+    cli.add_arg(parser, config, ["--docker_base_image"], default="ghcr.io/rjpower/levanter:latest")
+    cli.add_arg(parser, config, ["--docker_repository"], default="levanter")
+    cli.add_arg(parser, config, ["--foreground"], default=False, action="store_true")
+    cli.add_arg(parser, config, ["--image_name"], default=f"levanter-{getpass.getuser()}")
+    cli.add_arg(parser, config, ["--preemptible"], default=False, action="store_true")
+    cli.add_arg(parser, config, ["--project"], default=cli.gcloud_config()["project"])
+    cli.add_arg(parser, config, ["--tpu"], required=True)
```

Comment:
sorry to bikeshed, but can we call this 'name' maybe? I know `tpu_type` is right there, but it kind of looks like the kind of tpu you want